ui/abstract/abstract-multiple-edit-controller.js

```diff
@@ -0,0 +1,195 @@
+var Component = require("montage/ui/component").Component,
+    NotificationCenterModule = require("core/backend/notification-center");
+
+exports.AbstractMultipleEditController = Component.specialize({
+
+    _taskIds: {
+        value: null
+    },
+
+    initialize: {
+        value: function(sectionService, dataService) {
+            this._sectionService = sectionService;
+            this._dataService = dataService;
+
+            return this._load();
+        }
+    },
+
+    revert: {
+        value: function() {
+            return this._reset();
+        }
+    },
+
+    save: {
+        value: function() {
+            var self = this,
+                valuesMap = {},
+                promises = [];
+
+            this.values.forEach(function(value) {
+                if (!value.persistedId) {   // created
+                    promises.push(self._saveRawData(value));
+                } else {
+                    valuesMap[value.persistedId] = value;
+                }
+            });
+            
+            this._objects.forEach(function(object) {
+                var value = valuesMap[object.persistedId];
+                if (!value) { // deleted
+                    promises.push(self.deleteObject(object));
+                } else if (self.isValueUpdated(value, object)) { // updated
+                    promises.push(self._saveRawData(value));
+                }
+            });
+
+            return Promise.all(promises).then(function(taskIds) {
+                return self._startTaskDoneListener(taskIds);
+            });
+        }
+    },
+
+    handleObjectsRangeChange: {
+        value: function(plus, minus, index) {
+            return this._reset();
+        }
+    },
+
+    handleTaskDone: {
+        value: function(event) {
+            var taskId = event.detail.jobId;
+            if (this._taskIds && this._taskIds.length) {
+                this._taskIds.delete(taskId);
+                if (!this._taskIds.length) {
+                    this._reset();
+                    this._stopTaskDoneListener();
+                }
+            }
+        }
+    },
+
+    _load: {
+        value: function() {
+            var self = this;
+            return this.loadObjects().then(function(objects) {
+                self._objects = objects;
+                self._objects.addRangeChangeListener(self, "objects");
+
+                return self._reset();
+            });
+        }
+    },
+
+    _reset: {
+        value: function() {
+            this.values = this._objects.map(this.mapObjectToRawData);
+            return Promise.resolve();
+        }
+    },
+
+    _saveRawData: {
+        value: function(value) {
+            var self = this,
+                type = this.getModelType();
+
+            return this._dataService.getNewInstanceForType(type)
```

Comment:
Isn't there a way to avoid using `dataService` directly from the component ? It shouldn't be component's concern to directly save data.